samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -77,6 +77,11 @@
 TERRAFORM_BUILD_SCRIPT = "copy_terraform_built_artifacts.py"
 TF_BACKEND_OVERRIDE_FILENAME = "z_samcli_backend_override"
 
+HOOK_METADATA_KEY = "AWS::SAM::Hook"
```

Comment:
Do we need this constant defined here and in the utils fle?

Reply:
Well, my thinking is that if/when this was actually a hook package, it wouldn't have access to stuff in the sam cli repo/code. So I thought to code it as if it was a hook package. Not a big deal for me if we want to refer to the one in the utils or vice versa, however.